apps/server/src/modules/files-storage-client/uc/files-storage.producer.ts

```diff
@@ -0,0 +1,76 @@
+import { AmqpConnection } from '@golevelup/nestjs-rabbitmq';
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { RpcMessage } from '@shared/infra/rabbitmq/rpc-message';
+import { Logger } from '@src/core/logger';
+import {
+	FilesStorageEvents,
+	FilesStorageExchanges,
+	ICopyFileDO,
+	ICopyFilesOfParentParams,
+	IFileDO,
+	IFileRecordParams,
+} from '@src/shared/infra/rabbitmq';
+import { IFilesStorageClientConfig } from '../interfaces';
+import { ErrorMapper } from '../mapper/error.mapper';
+
+@Injectable()
+export class FilesStorageProducer {
+	private timeout = 0;
+
+	constructor(
+		private readonly logger: Logger,
+		private readonly amqpConnection: AmqpConnection,
+		private readonly configService: ConfigService<IFilesStorageClientConfig, true>
+	) {
+		this.logger.setContext(FilesStorageProducer.name);
+		this.timeout = this.configService.get('INCOMING_REQUEST_TIMEOUT');
+	}
+
+	async copyFilesOfParent(payload: ICopyFilesOfParentParams): Promise<ICopyFileDO[]> {
+		this.logger.debug({ action: 'copyFilesOfParent', payload });
+		const response = await this.amqpConnection.request<RpcMessage<ICopyFileDO[]>>({
```

Comment:
The response and creation is nearly the same for each case it is possible to put it to a private method that can handle all?
private createResponse(routingKey, payload): response